frontend/mobilityai/app/src/main/java/com/ai/mobility/mobilityai/MainActivity.java

```diff
@@ -599,19 +603,26 @@ private void startSyncing() {
         }
     }
 
-    private void addNewDevice(JSONObject response, String macAddr, int rssi) {
+    /**
+     * Add a new device to the device list after getting the device details from the server
+     * @param response Response containing first name, last name, and last synced date
+     * @param macAddr Mac address of new device to be added
+     * @param rssi Signal strength of new device to be added
+     */
+    private void addNewDevice(String response, String macAddr, int rssi) {
         String firstName = "Unregistered Device";
         String lastName = "";
         String lastSync = "Never";
 
         if(response != null) {
             try {
-                String id = response.getString("id");
+                JSONObject res = new JSONObject(response);
+                String id = res.getString("id");
 
                 if (!id.equals("-1")) {
```

Comment:
Can you change the API in the backend to return a http error code instead of -1 so we don't need to have this check on the android side?